workflows/src/main/kotlin/com/r3/corda/lib/tokens/workflows/internal/selection/VaultWatcherService.kt

```diff
@@ -124,35 +129,43 @@ class VaultWatcherService(tokenObserver: TokenObserver? = null) : SingletonSeria
         }
     }
 
+    // TODO Improve type safety for this method
     @Suspendable
     fun selectTokens(
             owner: Any,
-            amountRequested: Amount<IssuedTokenType>,
+            requiredAmount: Amount<TokenType>,
             predicate: ((StateAndRef<FungibleToken>) -> Boolean) = { true },
             allowShortfall: Boolean = false,
```

Comment:
does this param make sense? Why would one want to use it?

Reply:
Good question, I also agree that we shouldn't allow shortfall But as it is written in PR description it is refactor of in memory token selection as it was to be able to switch in by config. This is part of original code.